docs/source/plugin/test.rst

```diff
@@ -0,0 +1,323 @@
+
+If you have a custom section for your plugin, you will need to declare it, as
+you would do in your :func:`setup` function::
+
+    from your_plugin.config import MyPluginSection
+
+    def test_my_command(configfactory):
+        tmpconfig = configfactory(TEST_NAME, TEST_CONFIG)
+        tmpconfig.define_section('myplugin', MyPluginSection)
+
+And since we are using pytest, you can create your own local fixture for that::
+
+    @pytest.fixture
+    def tmpconfig(configfactory):
+        return configfactory(TEST_NAME, TMP_CONFIG)
+
+    def test_my_command(tmpconfig):
+        tmpconfig.define_section('myplugin', MyPluginSection)
+
+If all you need is a test configuration, you could put the section definition
+within your fixture. However, there are other options available to you when you
+use a test bot.
+
+Test bot
+--------
+
+Now that you have a test configuration available through your custom
+``tmpconfig`` fixture, you may want a test bot. Use the
+:func:`~sopel.tests.pytest_plugin.botfactory` fixture for that::
+
+    def test_my_command(tmpconfig, botfactory):
+        bot = botfactory(tmpconfig)
+
+However at this point, the bot doesn't know about your plugin, so it hasn't
+run the setup phase. You can do that with the
+:meth:`~sopel.tests.factories.BotFactory.preloaded` method::
+
+    def test_my_command(tmpconfig, botfactory):
+        bot = botfactory.preloaded(tmpconfig, ['myplugin'])
+        assert bot.has_plugin('myplugin')  # should be True
+
+.. note::
+
+    When using the :meth:`~sopel.tests.factories.BotFactory.preloaded` method,
+    you must not define your config sections manually, as this should be done
+    by your ``setup`` plugin hook.
+
+Of course, if you want to reuse the same test bot in all your tests, you can
+create a fixture for that::
+
+    @pytest.fixture
+    def bot(configfactory, botfactory):
+        settings = configfactory(TEST_NAME, TEST_CONFIG)
+        return botfactory.preloaded(settings, ['myplugin'])
+
+And then use it in your tests::
+
+    def test_my_command(bot):
+        assert bot.has_plugin('myplugin')  # should be True
+
+The ``bot`` created by the factory is a regular instance of
+:class:`sopel.bot.Sopel` with a
+:class:`test IRC backend <sopel.tests.mocks.MockIRCBackend>` instead of a
+regular one. This backend doesn't send anything over the network and instead
+it registers everything into its ``message_sent`` list::
+
+    from sopel.tests import rawlist
+
+
+    def test_my_command(bot):
+        bot.say('Hi!', '#channel')
+        assert len(bot.backend.message_sent) == 1
+        assert bot.backend.message_sent == rawlist(
+            'PRIVMSG #channel :Hi!',
+        )
+
+.. seealso::
+
+    For more information about the :func:`~sopel.tests.rawlist` function,
+    see the `Checking the bot's output`_ section.
+
+Test Server and test Users
+--------------------------
+
+Alright, you have a properly setup test bot, now it's time for the last parts
```

Comment:
```suggestion
Now that you have a properly setup test bot, it's time for the last parts
```

This is the writer in me making things flow better. I didn't really like the comma splice. x)

Reply:
Right now, I like the writer in you.